#PRIMER ALIGNEMNT WITH BLAST

This script perfromes an alignemtn of all primers listed in the primer database file agianst all sequences in the Sequences-folder with the BLAST algoritm. 

The alignment with the best BLAST-score is choosen for further downstream analysis where some basic calculations (Python-script) are perfromed and then vizulazed with R (se bellow for details).


#PYTHON SCRIPT

The Python script makes a data tabel consiting of several columns including the total mismatches (nucleotide mismatches, gaps and unaligned parts of the primer), a match precentage and aligment length. 

The script also discard primer aligments wich is not acclibale for the subtype (H1 primer are removed for H3 sequences etc.).


#R SCRIPT

The R script is just vizulazing the data tabel generated by the Python script. The heatmap shows the primer at x-axis and the sequences alignes at the y-axis. The fill of the heatmap is showing the numeber of total mismatches, while the number displayed in each square is the difference betweeen the lenght of the primer and the alignment (aligment differenc number, ADN). If the number displayed is higher than zero, the alignment is not aligned the whole primer while a number lower than zero indactes a gap or insertion. 

#INTERPATION

The interpetation of the heatmap is mostly straight forwars, but it is important to take the ADN into considertaion. A sample may have several mismatches in the heatmap, but if the ADN is high the mismatch number may be lower due to an unoptimazed alignemnt. Rule of thumb is then to alwasy take a closer look at samples with a high ADN before concluding. 






